applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -0,0 +1,464 @@
+    D(2, 3) = -(2 * mu * theta_new_b * (N_new(2) * N_new(3)));
+    D(2, 4) = -(2 * mu * theta_new_b * (N_new(2) * N_new(4)));
+    D(2, 5) = -(2 * mu * theta_new_b * (N_new(2) * N_new(5)));
+
+    D(3, 0) = -(2 * mu * theta_new_b * (N_new(3) * N_new(0)));
+    D(3, 1) = -(2 * mu * theta_new_b * (N_new(3) * N_new(1)));
+    D(3, 2) = -(2 * mu * theta_new_b * (N_new(3) * N_new(2)));
+    D(3, 3) = mu * theta_new - (2 * mu * theta_new_b * (N_new(3) * N_new(3)));
+    D(3, 4) = -(2 * mu * theta_new_b * (N_new(3) * N_new(4)));
+    D(3, 5) = -(2 * mu * theta_new_b * (N_new(3) * N_new(5)));
+
+    D(4, 0) = -(2 * mu * theta_new_b * (N_new(4) * N_new(0)));
+    D(4, 1) = -(2 * mu * theta_new_b * (N_new(4) * N_new(1)));
+    D(4, 2) = -(2 * mu * theta_new_b * (N_new(4) * N_new(2)));
+    D(4, 3) = -(2 * mu * theta_new_b * (N_new(4) * N_new(3)));
+    D(4, 4) = mu * theta_new - (2 * mu * theta_new_b * (N_new(4) * N_new(4)));
+    D(4, 5) = -(2 * mu * theta_new_b * (N_new(4) * N_new(5)));
+
+    D(5, 0) = -(2 * mu * theta_new_b * (N_new(5) * N_new(0)));
+    D(5, 1) = -(2 * mu * theta_new_b * (N_new(5) * N_new(1)));
+    D(5, 2) = -(2 * mu * theta_new_b * (N_new(5) * N_new(2)));
+    D(5, 3) = -(2 * mu * theta_new_b * (N_new(5) * N_new(3)));
+    D(5, 4) = -(2 * mu * theta_new_b * (N_new(5) * N_new(4)));
+    D(5, 5) = mu * theta_new - (2 * mu * theta_new_b * (N_new(5) * N_new(5)));
+}
+
+void LinearJ2Plasticity3D::GetLawFeatures(Features& rFeatures)
+{
+    rFeatures.mOptions.Set(THREE_DIMENSIONAL_LAW);
+    rFeatures.mOptions.Set(INFINITESIMAL_STRAINS);
+    rFeatures.mOptions.Set(ISOTROPIC);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Infinitesimal);
+    rFeatures.mStrainSize = 6;
+    rFeatures.mSpaceDimension = 3;
+}
+
+int LinearJ2Plasticity3D::Check(const Properties& rMaterialProperties,
+                                                 const GeometryType& rElementGeometry,
+                                                 const ProcessInfo& rCurrentProcessInfo)
+{
+    if(YOUNG_MODULUS.Key() == 0 || rMaterialProperties[YOUNG_MODULUS] <= 0.0) {
+        KRATOS_ERROR << "YOUNG_MODULUS has Key zero or invalid value " << std::endl;
+    }
+
+    const double& nu = rMaterialProperties[POISSON_RATIO];
+    const bool check = bool( (nu >0.499 && nu<0.501 ) || (nu < -0.999 && nu > -1.01 ) );
+    if(POISSON_RATIO.Key() == 0 || check) {
+        KRATOS_ERROR << "POISSON_RATIO has Key zero or invalid value " << std::endl;
+    }
+
+    if(YIELD_STRESS.Key() == 0 || rMaterialProperties[YIELD_STRESS] <= 0.0) {
+        KRATOS_ERROR << "YIELD_STRESS has Key zero or invalid value " << std::endl;
+    }
+
+    if(REFERENCE_HARDENING_MODULUS.Key() == 0) {
+        KRATOS_ERROR << "REFERENCE_HARDENING_MODULUS has Key zero" << std::endl;
+    }
+
+    if(ISOTROPIC_HARDENING_MODULUS.Key() == 0) {
+        KRATOS_ERROR << "ISOTROPIC_HARDENING_MODULUS has Key zero" << std::endl;
+    }
+
+    if(INFINITY_HARDENING_MODULUS.Key() == 0) {
```

Comment:
there are functions for this in kratos/includes/checks.h, can you please use them?